engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -373,6 +377,21 @@ object ProgramExecutionSupport {
           Api.ExpressionUpdate.Payload.DataflowError(
             ErrorResolver.getStackTrace(error).flatMap(_.expressionId)
           )
+        case withWarnings: WithWarnings =>
+          val warningsCount = withWarnings.getWarningsCount
+          val warning =
+            if (warningsCount == 1) {
+              val warnings = withWarnings.getWarningsArray(null)
+              Option(ctx.executionService.toDisplayString(warnings(0).getValue))
```

Comment:
Just to be on the safe side, can you guard that `warnings` has this single element?

Reply:
But we check `warningsCount == 1`

Reply:
OK, I somehow imagined that the count is set separately from the actual warnings because of the `Warnings` interface. All good.